webrender/src/prim_store.rs

```diff
@@ -1241,36 +1226,26 @@ impl PrimitiveStore {
             cpu_brushes: recycle_vec(self.cpu_brushes),
             cpu_text_runs: recycle_vec(self.cpu_text_runs),
 
-            pictures: recycle_vec(self.pictures),
-            next_picture_id: self.next_picture_id,
-
             chase_id: self.chase_id,
         }
     }
 
-    pub fn add_image_picture(
-        &mut self,
-        composite_mode: Option<PictureCompositeMode>,
-        is_in_3d_context: bool,
-        pipeline_id: PipelineId,
-        reference_frame_index: SpatialNodeIndex,
-        frame_output_pipeline_id: Option<PipelineId>,
-        apply_local_clip_rect: bool,
-    ) -> PictureIndex {
-        let picture = PicturePrimitive::new_image(
-            PictureId(self.next_picture_id),
-            composite_mode,
-            is_in_3d_context,
-            pipeline_id,
-            reference_frame_index,
-            frame_output_pipeline_id,
-            apply_local_clip_rect,
-        );
+    pub fn get_pic(&self, index: PrimitiveIndex) -> &PicturePrimitive {
```

Comment:
This looks like a regression in our internal APIs. Where we had separate type for picture index, we were able to associate it with a picture, now we pass more `PrimitiveIndex` around. I don't see a way around it if you want to move the picture under primitive metadata, but just pointing out at the issue.

Reply:
I agree - this is an internal regression. However, I think it should be temporary only - I'd like to embed more of the prim store types directly into the runs struct in the Picture (see rationale above) which should then remove (most of?) the indexing that occurs here, if all goes according to plan.